packages/salesforcedx-vscode-core/src/conflict/conflictDetectionService.ts

```diff
@@ -0,0 +1,218 @@
+      .withDescription(nls.localize('conflict_detect_convert_org_source'))
+      .withArg('force:mdapi:convert')
+      .withFlag('--rootdir', rootDir)
+      .withFlag('--outputdir', outputDir)
+      .withLogName('conflict_detect_convert_org_source')
+      .build();
+  }
+
+  public async checkForConflicts(
+    data: ConflictDetectionConfig
+  ): Promise<DirectoryDiffResults> {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const results = await this.checkForConflictsInternal(
+      getRootWorkspacePath(),
+      data,
+      cancellationTokenSource,
+      cancellationToken
+    );
+    return results;
+  }
+
+  private async checkForConflictsInternal(
+    projectPath: string,
+    data: ConflictDetectionConfig,
+    cancellationTokenSource: any,
+    cancellationToken: any
+  ): Promise<DirectoryDiffResults> {
+    const tempMetadataPath = this.getCachePath(data.usernameOrAlias);
+
+    // 1: create the shadow directory
+    const manifestPath = path.join(tempMetadataPath, 'package.xml');
+    try {
+      shell.mkdir('-p', tempMetadataPath);
+      shell.cp(data.manifest, manifestPath);
+    } catch (error) {
+      this.reportError('error_creating_packagexml', error);
+      return Promise.reject();
+    }
+
+    // 2: retrieve unmanaged org source to the shadow directory
+    await this.executeCommand(
+      this.buildRetrieveOrgSourceCommand(
+        data.usernameOrAlias,
+        tempMetadataPath,
+        manifestPath
+      ),
+      projectPath,
+      cancellationTokenSource,
+      cancellationToken
+    );
+
+    // 3: unzip retrieved source
+    const unpackagedZipFile = path.join(tempMetadataPath, 'unpackaged.zip');
+    try {
+      const zip = new AdmZip(unpackagedZipFile);
+      zip.extractAllTo(tempMetadataPath, true);
+    } catch (error) {
+      this.reportError('error_extracting_org_source', error);
+      return Promise.reject();
+    }
+
+    // 4: convert org source to decomposed (source) format
+    const unconvertedSourcePath = path.join(tempMetadataPath, 'unpackaged');
+    const convertedSourcePath = path.join(tempMetadataPath, 'converted');
+    await this.executeCommand(
+      this.buildMetadataApiConvertOrgSourceCommand(
+        unconvertedSourcePath,
+        convertedSourcePath
+      ),
+      projectPath,
+      cancellationTokenSource,
+      cancellationToken
+    );
+
+    // 5: diff project directory (local) and retrieved directory (remote)
+    // Assume there are consistent subdirs from each root i.e. 'main/default'
+    const localSourcePath: string = path.join(projectPath, data.packageDir);
+    const diffs = this.differ.diff(localSourcePath, convertedSourcePath);
+
+    // 6: cleanup temp directory
+    try {
+      shell.rm('-rf', tempMetadataPath);
+    } catch (error) {
+      // Failed cleanup should not terminate the service
+      this.reportError('error_cleanup_temp_files', error);
```

Comment:
In case this fails to clean up the temp folder, is there a situation where the user could run into false positives caused by old files being around ? I'm thinking we might want to check if the tmp directory is present and delete it before unzipping & converting the code.

Reply:
It will produce false positives, I'll add the cleanup up stream.